--- a/libhttpd/protocol.c
+++ b/libhttpd/protocol.c
@@ -671,19 +671,19 @@
 #define URL_XALPHAS     (unsigned char) 1
 #define URL_XPALPHAS    (unsigned char) 2
 
-/*      Bit 0           xalpha          -- see HTFile.h
+/*      Bit 0           xalpha          -- see RFC 1630
 **      Bit 1           xpalpha         -- as xalpha but with plus.
 **      Bit 2 ...       path            -- as xpalpha but with /
 */
     /*   0 1 2 3 4 5 6 7 8 9 A B C D E F */
-    {    7,0,0,0,0,0,0,0,0,0,7,0,0,7,7,7,       /* 2x   !"#$%&'()*+,-./ */
+    {    0,7,7,0,7,0,7,7,7,7,7,6,7,7,7,4,       /* 2x   !"#$%&'()*+,-./ */
          7,7,7,7,7,7,7,7,7,7,0,0,0,0,0,0,       /* 3x  0123456789:;<=>?  */
          7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,       /* 4x  @ABCDEFGHIJKLMNO */
          7,7,7,7,7,7,7,7,7,7,7,0,0,0,0,7,       /* 5X  PQRSTUVWXYZ[\]^_ */
          0,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7,       /* 6x  `abcdefghijklmno */
          7,7,7,7,7,7,7,7,7,7,7,0,0,0,0,0 };     /* 7X  pqrstuvwxyz{\}~ DEL */
  
-#define ACCEPTABLE(a)   ( a>=32 && a<128 && ((isAcceptable[a-32]) & mask))
+#define ACCEPTABLE(a)   ( a!='&' && a>=32 && a<128 && ((isAcceptable[a-32]) & mask))
 
 static char *hex = "0123456789ABCDEF";
 
